openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/CollectionDAO.java

```diff
@@ -442,6 +442,170 @@ default Class<Container> getEntityClass() {
+    }
+
+    @Override
+    default List<String> listAfter(ListFilter filter, int limit, String after) {
+      boolean root = Boolean.parseBoolean(filter.getQueryParam("root"));
+      String condition = filter.getCondition();
+
+      if (!root) {
+        return EntityDAO.super.listAfter(filter, limit, after);
+      }
+
+      StringBuilder mysqlCondition = new StringBuilder();
+      StringBuilder psqlCondition = new StringBuilder();
+
+      mysqlCondition.append(String.format("%s ", condition));
+      psqlCondition.append(String.format("%s ", condition));
+
+      mysqlCondition.append("AND er.toId is NULL");
+      psqlCondition.append("AND er.toId is NULL");
+
+      return listAfter(
+          getTableName(), getNameColumn(), mysqlCondition.toString(), psqlCondition.toString(), limit, after);
+    }
+
+    @Override
+    default int listCount(ListFilter filter) {
+      boolean root = Boolean.parseBoolean(filter.getQueryParam("root"));
+      String condition = filter.getCondition();
+
+      if (!root) {
+        return EntityDAO.super.listCount(filter);
+      }
+
+      StringBuilder mysqlCondition = new StringBuilder();
+      StringBuilder psqlCondition = new StringBuilder();
+
+      mysqlCondition.append(String.format("%s ", condition));
+      psqlCondition.append(String.format("%s ", condition));
+
+      mysqlCondition.append("AND er.toId is NULL");
+      psqlCondition.append("AND er.toId is NULL");
+
+      return listCount(getTableName(), getNameColumn(), mysqlCondition.toString(), psqlCondition.toString());
+    }
+
+    @ConnectionAwareSqlQuery(
+        value =
+            "SELECT json FROM ("
+                + "SELECT <nameColumn>, ce.json FROM <table> ce "
+                + "LEFT JOIN ("
+                + "  SELECT toId FROM entity_relationship "
+                + "  WHERE fromEntity = 'container' AND toEntity = 'container' AND relation = 0 "
+                + ") er "
+                + "on ce.id = er.toId "
+                + "<mysqlCond> AND "
+                + "<nameColumn> < :before "
+                + "ORDER BY <nameColumn> DESC "
+                + "LIMIT :limit"
+                + ") last_rows_subquery ORDER BY <nameColumn>",
+        connectionType = MYSQL)
+    @ConnectionAwareSqlQuery(
+        value =
+            "SELECT json FROM ("
+                + "SELECT <nameColumn>, ce.json FROM <table> ce "
+                + "LEFT JOIN ("
+                + "  SELECT toId FROM entity_relationship "
+                + "  WHERE fromEntity = 'container' AND toEntity = 'container' AND relation = 0 "
+                + ") er "
+                + "on ce.id = er.toId "
+                + "<psqlCond> AND "
+                + "<nameColumn> < :before "
+                + "ORDER BY <nameColumn> DESC "
+                + "LIMIT :limit"
+                + ") last_rows_subquery ORDER BY <nameColumn>",
+        connectionType = POSTGRES)
```

Comment:
nit: If they are the same, can we use `@SqlQuery`? Same for other places where this comment applies.

Reply:
done